pkgs/containers/nvidia-ctk.nix

```diff
@@ -0,0 +1,38 @@
+{ fetchFromGitLab, buildGoModule, fetchpatch2 }:
+
+buildGoModule rec {
+  pname = "nvidia-ctk";
+  version = "unstable-${builtins.substring 0 7 src.rev}";
+
+  src = fetchFromGitLab {
+    owner = "nvidia";
+    repo = "container-toolkit/container-toolkit";
+    rev = "a2262d00cc6d98ac2e95ae2f439e699a7d64dc17";
+    hash = "sha256-Oi04PIES0qTih/EiFBStIoBadM3H52+81KEfUumQcIs=";
+  };
+
+  patches = [
+    (fetchpatch2 {
```

Comment:
Do we need fetchpatch2? It has consistency problems as we've seen with edk2.

Reply:
Also, could you add comments about why we need these patches?

Reply:
fetchpatch2 isn't strictly needed, I switched it to fetchpatch. I'll need to brush up on the differences between the two, I thought for most use cases fetchpatch2 still handled things better than fetchpatch